frontend/packages/react-app/src/components/pages/ExplorePage.tsx

```diff
@@ -22,26 +58,29 @@ const ExplorePage = () => {
   useEffect(() => {
     if (walletAddress !== "") {
       getTokensBalance({
-        variables: {id: walletAddress.toLowerCase()}
-      })
+        variables: { id: walletAddress.toLowerCase() },
+      });
     }
   }, [walletAddress, getTokensBalance]);
 
   useEffect(() => {
     if (loading || error || !data) {
-      return
+      return;
     }
     if (!data.account) {
-      return
+      return;
     }
-    const tmpTokens = data.account.tokens.map(accountToken => ({
-        address: accountToken.token.id,
-        name: accountToken.token.name,
-        symbol: accountToken.token.symbol,
-        balance: ethers.utils.formatUnits(accountToken.balance, accountToken.token.decimals)
-    }))
-    setTokens(tmpTokens)
-  }, [loading, error, data, setTokens]);
+    const tmpTokens = data.account.tokens.map((accountToken) => ({
+      address: accountToken.token.id,
+      name: accountToken.token.name,
+      symbol: accountToken.token.symbol,
+      balance: ethers.utils.formatUnits(
+        accountToken.balance,
+        accountToken.token.decimals
+      ),
+    }));
+    dispatch({ type: ACTIONS.SET_TOKENS, payload: { tokens: tmpTokens } });
```

Comment:
According to redux style guide, putting a logic into reducer is recommended.
So we can rewrite like following: 

```javascript
function reducer(state: ExplorePageState, action: ExplorePageAction) {
  switch (action.type) {
    (...)
    case ACTIONS.SET_TOKENS:
      if (!action.payload.data.account) {
        return state
      }
      const tokens = action.payload.data.account.tokens.map((accountToken) => ({
        address: accountToken.token.id,
        name: accountToken.token.name,
        symbol: accountToken.token.symbol,
        balance: ethers.utils.formatUnits(
            accountToken.balance,
            accountToken.token.decimals
        ),
      }));
      return { ...state, tokens: [...tokens] };
    (...)
  }
}

const ExplorePage = () => {
  (...)
  useEffect(() => {
    if (loading || error || !data) {
      return;
    }
    dispatch({ type: ACTIONS.SET_TOKENS, payload: { data } });
  }, [loading, error, data]);
  (...)
} 
```

https://redux.js.org/style-guide/style-guide#put-as-much-logic-as-possible-in-reducers
https://qiita.com/nouka/items/74a2784598e7bd5554cd

Reply:
OK!